[WIP] Add documentation for Seccomp GA (#21278)
kubernetes/website: content/en/docs/concepts/policy/pod-security-policy.md

### Seccomp

> You can use annotations on Pods to control use of seccomp profiles, and you can
> use PodSecurityPolicies to enforce how these annotations are applied.

Reviewer: May be worth focusing on secomp GA behaviour:

```suggestion
You can use the seccompProfile field on Pods or containers to control use of seccomp profiles, 
and you can use PodSecurityPolicies to enforce how these fields are applied.
```

### Seccomp

`seccompProfile` contains two fields that specify the seccomp profile that will

> be used for the pod.

Reviewer 2: I suggest that we add a link to the new seccomp site here, too.

> be used for the pod.

Author: @pjbgf  Ah great point! I misread what was being described here :/ I'll revert changes and remove the alpha specification :+1:

> be used for the pod.

Author: @saschagrunert by site do you mean the full guide that will be in `/tasks`?

> be used for the pod.

Reviewer 2: > @saschagrunert by site do you mean the full guide that will be in `/tasks`?

Yes :+1: :)

**Type** - determines the seccomp profile type for the pod

- `Unconfined` - Seccomp is not applied to the container processes (this is the
  default in Kubernetes), if no alternative is provided.
- `RuntimeDefault` - The default container runtime profile is used.
- `Localhost` - Specify a seccomp profile as a file on the node. Must set
  `LocalhostProfile` if this option is provided.

**LocalhostProfile** - Path of the seccomp profile in the format
  `<seccomp_root>/<path>`, where `<seccomp_root>` is defined via the

>   `--seccomp-profile-root` flag on the Kubelet.

Reviewer: It may be worth mentioning that the flag `--seccomp-profile-root` is now deprecated and is due to be removed on v1.23.

- `docker/default` - The Docker default seccomp profile is used. Deprecated as of
  Kubernetes 1.11. Use `runtime/default` instead.
- `localhost/<path>` - Specify a profile as a file on the node located at
- `RuntimeDefault` - The default container runtime profile is used.
- `Localhost` - Specify a seccomp profile as a file on the node. Must set
  `LocalhostProfile` if this option is provided.

**LocalhostProfile** - Path of the seccomp profile in the format
  `<seccomp_root>/<path>`, where `<seccomp_root>` is defined via the
  `--seccomp-profile-root` flag on the Kubelet.

**seccomp.security.alpha.kubernetes.io/allowedProfileNames** - Annotation that
specifies which values are allowed for the pod seccomp annotations. Specified as
a comma-delimited list of allowed values. Possible values are those listed
above, plus `*` to allow all profiles. Absence of this annotation means that the
default cannot be changed.

> 

Reviewer: This will need to be reverted.

### Seccomp

You can use the `seccompProfile` field in the `securityContext` of Pods or
containers to [control use of seccomp
profiles](../../tutorials/clusters/seccomp.md), and you can use

> PodSecurityPolicies to enforce how these fields are applied.

Reviewer 2: Let's mention that the PSP enforces both: the annotation and the new fields like described here: https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/20190717-seccomp-ga.md#podsecuritypolicy-enforcement

Author: Great point, thanks @saschagrunert!

Author: Updated :+1:

> The use of seccomp profiles in pods can be controlled via annotations on the

Reviewer 3: ```suggestion
You can use annotations on Pods to control use of seccomp profiles, and you can use
PodSecurityPolicies to enforce how these annotations are applied.
```

- `unconfined` - Seccomp is not applied to the container processes (this is the
  default in Kubernetes), if no alternative is provided.
- `runtime/default` - The default container runtime profile is used.
- `docker/default` - The Docker default seccomp profile is used. Deprecated as
  of Kubernetes 1.11. Use `runtime/default` instead.
- `localhost/<path>` - Specify a profile as a file on the node located at
  `<seccomp_root>/<path>`, where `<seccomp_root>` is defined via the

>   `--seccomp-profile-root` flag on the Kubelet. (Note: this flag has been
>   deprecated in v1.19 and will be removed in v1.23. The seccomp root path will
>   then be derived from the kubelet root path, which is defined by `--root-dir`.
>   The current default value is `<root-dir>/seccomp`.)

Reviewer 3: ```suggestion
  `--seccomp-profile-root` flag on the kubelet.

   {{< note >}}
   Kubernetes v1.19 deprecated the `--seccomp-profile-root` flag, which will be removed in
   Kubernetes v1.23. The seccomp root path will   then be derived from the kubelet root path,
   which is defined by `--root-dir`. The current default value is `<root-dir>/seccomp`.
   {{< /note >}}
```
_or_, with [avoid statements about the future](https://kubernetes.io/docs/contribute/style/style-guide/#avoid-statements-about-the-future) in mind:
```suggestion
   the seccomp root path for that node. The default seccomp root path is a directory
   named `seccomp` directly beneath the kubelet's root directory (specified via the
   `--root-dir` command line flag to the kubelet).

   {{< note >}}
   The `--seccomp-profile-root` flag (deprecated since Kubernetes v1.19) allowed you to configure 
   the seccomp root path for a node. To use the defined behavior, leave that flag unset so that the
   kubelet uses its default.   
   {{< /note >}}
```

Reviewer 4: @hasheddan , avoid statements/promises/predictions about the future.
